### DISCUSSION TITLE
New AD5X

### DISCUSSION THREAD

**Discussion by MACCAMENZIES** [xblax/flashforge_ad5m_klipper_mod#259] [General]
It's been leaked 
Would imagine it has more memory on board now that it's running 4 colors![image](https://github.com/user-attachments/assets/035b41b4-99f5-4f4f-b026-313041f2641b)

**Comment by KaruroChori** [comment]
Not really, changes in filament are defined by the slicer and are just basic, slow gcode moves. There is no real difference in the workload performed by MCUs and main board.

**Reply by MACCAMENZIES** [reply]
So as far as our kipper mod will the firmware have to be changed ?

**Reply by KaruroChori** [reply]
Best option: they ship a newer version of klipper with this revised printer; the hardware is still mostly the same, and there is no meaningful change in the architecture of their software. In this case we should be able to backport the new klipper to the ad5m/pro as well and we all get to live happy ever since.

Likely option: the "new" firmware is still based on the same old branch of klipper; as it is, the printer can already work with multiple materials; there should be only minor changes to the configuration files and macros to support the new mcu which switches filaments. Most of them would be copied and pasted from the original files from flashforge.

Worst case scenario: new incompatible hardware or some custom protocol to handle the filament changer. In that case major changes would be needed, and modding might even not be possible as we are currently doing it. Even worse, none of that effort would improve the current ad5m/pro printers.

**Reply by MACCAMENZIES** [reply]
Fingers crossed.

**Reply by mircsicz** [reply]
+1, I Hope someone gets his Fingers on a Unit for tests...

**Reply by max-hasky** [reply]
AD5X support is in [zmod](https://github.com/ghzserg/zmod).
KlipperMod can also be ported to AD5X, but you need to rebuild BuildRoot for mips and change the paths.

**Reply by TMTYD** [reply]
Hi

Why doesn't Klipperscreen work on the AD5x?

**Reply by guitarguygizmo** [reply]
> AD5X support is in [zmod](https://github.com/ghzserg/zmod). KlipperMod can also be ported to AD5X, but you need to rebuild BuildRoot for mips and change the paths.

Do you know if there's a guide on this process anywhere? I'm trying to do so myself to be able to speed up the color change process even if just a bit. I tried looking for something but all google is doing is bringing me back to your comment lol

**Reply by max-hasky** [reply]
I don't know how to do this on KlipperMod, but [work is underway to speed up filament change](https://github.com/ghzserg/zmod/wiki/AD5X_en#6-how-to-configure-waste-filament-during-filament-change).

**Reply by xblax** [reply]
AD5X is not supported by Klipper Mod. Sorry, non of the developers has that printer.

**Reply by Plastic-armyman** [reply]
> > AD5X support is in [zmod](https://github.com/ghzserg/zmod). KlipperMod can also be ported to AD5X, but you need to rebuild BuildRoot for mips and change the paths.
> 
> Do you know if there's a guide on this process anywhere? I'm trying to do so myself to be able to speed up the color change process even if just a bit. I tried looking for something but all google is doing is bringing me back to your comment lol

there are instructions on the github
https://github.com/ghzserg/zmod
